app/src/main/java/com/kazakov/newyou/app/view/component/base/impl/view/PredictionView.java

```diff
@@ -41,9 +61,188 @@ public boolean onOptionsItemSelected(MenuItem item) {
         return true;
     }
 
-    private void renderView(List<PredictionResult> predictionResults) {
-        TextView textView = new TextView(getActivity());
-        textView.append("hello");
-        currentFrame.addView(textView);
+    private void renderView(List<PredictionResult> predictionResult) {
+        setToggleButtonListener(predictionResult);
+        tableLayout.setShrinkAllColumns(true);
+        for (PredictionResult p : predictionResult) {
+            TableRow tableRow = addPredictionToRow(p);
+            tableLayout.addView(tableRow);
+        }
+    }
+
+    private TableRow addPredictionToRow(PredictionResult p) {
+        TableRow tableRow = new TableRow(tableLayout.getContext());
+        createActivityInfoCell(p.getActivity().toString(), tableRow);
+        tableRow.addView(getDisabledTextViewForTableRow(p.getDuration(), tableRow));
+        createNumberOfRepeatCell(Integer.toString(p.getNumber_of_repeats()), tableRow);
+        return tableRow;
+    }
+
+    private TextView getDisabledTextViewForTableRow(String text, TableRow row) {
+        TextView editText = new TextView(row.getContext());
+        editText.setText(text);
+        editText.setEnabled(false);
+        editText.setWidth(cellSize);
+        editText.setTextAlignment(View.TEXT_ALIGNMENT_CENTER);
+        editText.setTextColor(Color.BLACK);
+        return editText;
+    }
+
+    private void createActivityInfoCell(String text, TableRow row) {
+
+        String[] texts = activityToStringArray();
+        TextSwitcher simpleTextSwitcher = createSimpleTextSwitcher(row.getContext());
+
+        simpleTextSwitcher.setCurrentText(text);
+        Button btnNext = new Button(row.getContext());
+        btnNext.setEnabled(false);
+        btnNext.setBackgroundResource(R.drawable.turn_right);
+        btnNext.setOnClickListener(v -> moveTextRight(texts, simpleTextSwitcher));
+        Button btnPrev = new Button(row.getContext());
+        btnPrev.setBackgroundResource(R.drawable.turn_left);
+
+        btnPrev.setOnClickListener(v -> moveTextLeft(texts, simpleTextSwitcher));
+        row.addView(btnPrev);
+        row.addView(simpleTextSwitcher);
+        row.addView(btnNext);
+        btnPrev.setVisibility(View.INVISIBLE);
+        btnNext.setVisibility(View.INVISIBLE);
+
+    }
+
+    private void createNumberOfRepeatCell(String text, TableRow row) {
```

Comment:
Is it possible to decompose this class into creational logic using factories etc and behavior logic using activity itself?